torch/onnx/symbolic.py

```diff
@@ -80,7 +80,13 @@ def _parse_arg(value, desc):
         elif desc == 't':
             return tval
         elif desc == 'is':
-            return [int(v) for v in tval]
+            try:
+                return [int(v) for v in tval]
+            except TypeError:
+                if tval.dim():
+                    raise
+                # scalar tensors are not iterable
+                return [int(tval)]
```

Comment:
Hi @barrh, this is not the right way to handle different type of inputs. You can use instance() function or _is_value + lengths.type().isSubtypeOf(torch._C.TensorType.get()). There are couples of examples in symbolic.py. Also, please add an end to end test case to test_pytorch_onnx_caffe2.py. We don't encourage adding more test cases to test_operators.py if unnecessary.

Reply:
@houseroad  I didn't understand your comment regarding the handling.
To allow this handling scale beyond 1 element, I iterate the tensor, but catch exception for scalar tensors (e.g. `torch.tensor(1)`. The TypeError is raised due to the traversing on non-iterable, not the element's type.